library_management_cookiecutter/library/views.py

```diff
@@ -0,0 +1,243 @@
+from django import http
+from django.shortcuts import render
+from django.shortcuts import redirect
+from django.views.generic import TemplateView, CreateView, ListView,FormView
+from .models import User, Book,AssignedBook
+from .forms import UserForm, AddBook,  AsignBook
+from django.contrib.auth.views import LoginView, LogoutView
+from django.views import View
+from django.http import HttpRequest, HttpResponse, JsonResponse
+from django.urls import reverse, reverse_lazy
+from django.contrib.auth import authenticate, login
+from django.contrib import messages
+from django.contrib.auth.mixins import LoginRequiredMixin
+from .mixin import MyCustomPermissions
+from django.contrib.auth.models import Group, Permission
+from django.core import serializers
+from django.conf import settings
+from django.db.models import Count
+from django.utils import timezone
+import csv
+from django.core.mail import EmailMessage
+import zipfile
+import os
+import pyminizip
+from io import StringIO
+
+class Home(TemplateView):
+    template_name = "home.html"
+
+
+class Login(LoginView):
+    """login class"""
+
+    template_name = "login.html"
+
+    def post(self, request, *args, **kwargs):
+        username = request.POST.get("username", None)
+        password = request.POST.get("password", None)
+        user = authenticate(username=username, password=password)
+        if user is not None:
+            login(request, user)
+            return JsonResponse({"message": "success"})
+        return JsonResponse({"message": "username and password not match."}, status=400)
+
+
+class Logout(LogoutView):
+    """logout class"""
+
+    pass
+
+
+class CreateUser(CreateView):
```

Comment:
why are you using POst method inside create view ? because create view also create the data. why you need to write extra ine of code..

Reply:
because user will select role automatically then group was selected . if group not selected that time  i can not apply user permissions so use post method.